agent/lib/kontena/workers/stats_worker.rb

```diff
@@ -30,6 +30,9 @@ def configure_statsd(node)
       else
         @statsd = nil
       end
+    rescue => error
+      warn "statsd configuration failed: #{error.message}"
+      warn error.backtrace.join('\n') if error.backtrace
```

Comment:
To be able to figure that out, I think it would need to dive into the `SocketError` message details. Doing logic based on exception messages is always a bit of a rabbit hole... :D